### PR TITLE
health.vim: various fixes

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: runtime/autoload/health/provider.vim [hunk 1]
```diff
@@ -38,6 +38,16 @@ function! s:system_handler(jobid, data, event) dict abort
...
 endfunction
 
+" Attempts to construct a shell command from an args list.
+" Only for display, to help users debug a failed command.
+function! s:shellify(cmd) abort
+  if type(a:cmd) != type([])
+    return a:cmd
+  endif
+  return join(map(a:cmd,
+    \'v:key == 0 || v:val !~# ''[^\-.a-zA-Z_/]'' ? v:val : shellescape(v:val)'), ' ')
```
Comment (ZyX-I) [inline runtime/autoload/health/provider.vim]: Checking for `v:key == 0` does not seem right, it is valid to have paths with e.g. spaces. Especially on cygwin. Second condition with double negation is hard to parse. I would simply write `v:val =~# ''\m[\-.a-zA-Z_/]'' ? shellescape(v:val) : v:val`. (`\m` is a convention raising from “I do not want to remember where &magic applies and where it does not”.)

File: runtime/autoload/health/provider.vim [hunk 2]
```diff
@@ -38,6 +38,16 @@ function! s:system_handler(jobid, data, event) dict abort
   endif
 endfunction
 
+" Attempts to construct a shell command from an args list.
+" Only for display, to help users debug a failed command.
+function! s:shellify(cmd) abort
+  if type(a:cmd) != type([])
+    return a:cmd
+  endif
+  return join(map(a:cmd,
```
Comment (ZyX-I) [inline runtime/autoload/health/provider.vim]: And `map(copy(a:cmd)`. Or you may run into issues some day.

**Comment by vext01** [comment]
Thanks!
